src/main/java/org/broadinstitute/hellbender/tools/exome/ExonCollections.java

```diff
@@ -0,0 +1,108 @@
+package org.broadinstitute.hellbender.tools.exome;
+
+import htsjdk.tribble.FeatureCodec;
+import htsjdk.tribble.bed.BEDFeature;
+import org.broadinstitute.hellbender.engine.FeatureDataSource;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.io.File;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+/**
+ * Utility class used to create {@link ExonCollections} from common exon data sources.
+ *
+ * @author Valentin Ruano-Rubio &lt;valentin@broadinstitute.org&gt;
+ */
+public final class ExonCollections {
+
+    /**
+     * Disabled instantiation of this helper class.
+     */
+    private ExonCollections() {
+
+    }
+
+    /**
+     * Creates an exon collection out from an interval list where each element corresponds to an exon.
+     *
+     * @param intervals the exon intervals.
+     * @throws IllegalArgumentException if {@code intervals} is {@code null}, it contains the {@code null} element or
+     *   overlapping intervals.
+     * @return never {@code null}.
+     */
+    public static ExonCollection<SimpleInterval> fromSimpleIntervalList(final List<SimpleInterval> intervals) {
+        return new HashedListExonCollection<SimpleInterval>(intervals) {
+
+            @Override
+            public String name(final Object exon) {
+                Utils.nonNull(exon,"the input exon cannot be null");
+                if (exon instanceof SimpleInterval) {
```

Comment:
It just to be so, but there is a java Generics situation that render it problematic without casting and SupressingWarnings somewhere else ... 

It has to do with the fact that ExonReadCounts.exons is ```ExonCollection<? extends Locatable>``` to accommodate for the possibility of SimpleIntervals or BEDFeature as possible element types. 

It would be similar to ```Map.get(Object)``` or ```equals(Object)```. I will double check whether it can be avoided but I won't promise. 

In any case the contract of ```name(Object)``` and ```location(Object)``` allowed for this before this pull-request.